web_field_company_dependent_flag/__init__.py

```diff
@@ -0,0 +1,12 @@
+from odoo.tools import config
+from .hooks import pre_init_partner_phone
+
+from . import models
+
+if not config.get("without_demo"):
+    from . import demo
```

Comment:
I' m not sure adding demo python file is a good thing. I think we could use the classic odoo_test_helper tools.
See for exemple : 
- https://github.com/OCA/server-tools/blob/18.0/base_exception/tests/test_base_exception.py#L5
- https://github.com/OCA/server-tools/blob/18.0/base_time_window/tests/test_time_window_mixin.py#L1
- https://github.com/OCA/stock-logistics-barcode/blob/16.0/barcodes_generator_abstract/tests/test_barcodes_generator_abstract.py#L4